### PR TITLE
Add online reader

### PR DESCRIPTION
**Pull request by NicholasAntonov**

### CODE REVIEW

File: frontend/templates/read.html [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{% load sass_tags %}
+
+<!DOCTYPE html>
+<head>
+    <title>Read {{ work.title }} online at unglue.it</title>
```
Comment (eshellman) [inline frontend/templates/read.html]: I assume this head is mostly the same as on work page? If so, we'll want to extract the common code into one template.

File: frontend/templates/work.html [hunk 2]
```diff
@@ -179,11 +179,20 @@ <h4>Related Works</h4>
             </div>
         </div>
         <div class="bookSidebar">
+            <div class="bookRead">
```
Comment (eshellman) [inline frontend/templates/work.html]: We're going to need some logic to suppress the button if no usable ebook. Same with donate and download buttons

File: frontend/views/__init__.py [hunk 3]
```diff
@@ -297,8 +297,21 @@ def stub(request):
...
+    except (ValueError, AttributeError):
+        raise Http404
+    return render(request, 'read.html', {
+        'work': work,
+        'url': url,
+    })
+
 def work(request, work_id, action='display'):
     work = safe_get_work(work_id)
+    print work.first_epub().id
```
Comment (eshellman) [inline frontend/views/__init__.py]: debug code?

File: frontend/templates/read.html (hunk 1 above)
Comment (NicholasAntonov) [reply frontend/templates/read.html]: Unfortunately it is mostly different. the only common code is the meta tags, which we could extract away.

Comment (eshellman) [reply frontend/templates/read.html]: that's a should do, then

File: frontend/views/__init__.py [hunk 4]
```diff
@@ -422,7 +422,8 @@ def work(request, work_id, action='display'):
         'cover_width': cover_width_number,
         'action': action,
         'formset': formset,
-        'kwform': SubjectSelectForm()
+        'kwform': SubjectSelectForm(),
+        'has_online_book': work.first_epub() != None,
```
Comment (eshellman) [inline frontend/views/__init__.py]: look at the code for models.Ebook.get_archive_ebf  You want to make sure `ebook.provider in good_providers` and then use archive_ebf as the epub source

**Pull request merged**